hs_collection_resource/page_processors.py

```diff
@@ -0,0 +1,176 @@
+        viewable_resources = user.uaccess.get_resources_with_explicit_access(PrivilegeCodes.VIEW)
+
+        owned_resources = list(owned_resources)
+        editable_resources = list(editable_resources)
+        viewable_resources = list(viewable_resources)
+        discovered_resources = list(user.ulabels.my_resources)
+
+        user_all_accessible_resource_list = (owned_resources + editable_resources + \
+                                             viewable_resources + discovered_resources)
+    else: # anonymous user
+        user_all_accessible_resource_list = list(BaseResource.objects. \
+                                                 filter(Q(raccess__public=True) | Q(raccess__discoverable=True)).distinct())
+
+    collection_items_list = None
+    collection_items_accessible = []
+    collection_items_inaccessible = []
+    if content_model.metadata.collection_items.first():
+        collection_items_list = list(content_model.metadata.collection_items.first().collection_items.all())
+        for res in collection_items_list:
+            if res in user_all_accessible_resource_list or res.raccess.discoverable or res.raccess.public:
+                collection_items_accessible.append(res)
+            else:
+                collection_items_inaccessible.append(res)
+
+    if not edit_resource:
+        # get the context from hs_core
+        context = page_processors.get_page_context(page, request.user,
+                                                   resource_edit=edit_resource,
+                                                   extended_metadata_layout=None,
+                                                   request=request)
+        extended_metadata_exists = False
+
+        collection_message = ""
+        if collection_items_list is not None:
+            collection_length = len(collection_items_list)
+            collection_message = "This collection holds {0} resource(s) in total. ".format(collection_length)
+            hide_count = len(collection_items_inaccessible)
+            if hide_count > 0:
+                collection_message += "You have NO permission to view the {} grey resource(s). \
+                You may request for permission by leaving a comment.".format(hide_count)
+
+        if collection_items_list is not None:
+            extended_metadata_exists = True
+
+            link_html = ""
+            for res in collection_items_accessible:
+                sharing_status = get_sharing_status_string(res)
+                link_html += '<a style="color: blue" href="/resource/' + res.short_id + '" target="_blank">' + \
+                             res.resource_type + ' : ' + res.title + ' : ' + sharing_status + ' : ' + \
+                             res.short_id + '</a><br/>'
+            for res in collection_items_inaccessible:
+                sharing_status = get_sharing_status_string(res)
+                link_html += '<a style="color: grey" href="/resource/' + res.short_id + '" target="_blank">' + \
```

Comment:
@pkdash  we have some unit tests, see hs_collection_resource/tests/test_collection.py. Will update them once we make major changes. thanks